<?xml version="1.0"?>
<launch>
	<!-- ros loop rate -->
    <arg name="hz" default="100.0"/>
	<!-- grid -->
    <arg name="width" default="10.0"/>
    <arg name="grid_num" default="50"/>
	<!-- flow -->
    <arg name="manual_crop_size" default="5"/>
    <arg name="is_sparce" default="true"/>
    <arg name="max_corners" default="20"/>
    <arg name="quality_level" default="0.05"/>
    <arg name="min_distance" default="5.0"/>
    <arg name="win_size" default="10"/>
    <arg name="max_count" default="5"/>
    <arg name="min_human_velocity" default="0.0"/>
    <arg name="max_human_velocity" default="1.0"/>
	<!-- frame -->
    <arg name="frame_id" default="odom"/>
    <arg name="child_frame_id" default="base_footprint"/>
	<!-- sim or bag -->
    <!-- <arg name="is_gazebo" default="true"/> -->

	<!-- dynamic_cloud_detector -->
    <arg name="resolution" default="0.20"/>
    <arg name="occupancy_threshold" default="0.3"/>
    <arg name="beam_num" default="720"/>
    <arg name="skip_scan_flag" default="false"/>


	<param name="use_sim_time" value="false"/>

	<!-- <node pkg="bev_converter" type="ground_truth_odom_publisher" name="ground_truth_odom_publisher"> -->
	<!-- </node> -->
	<node pkg="bev_converter" type="raycast_publisher" name="raycast_publisher" output="screen">
		<!-- ros loop rate -->
		<param name="Hz" value="$(arg hz)" type="double"/>
		<!-- grid -->
		<param name="WIDTH" value="$(arg width)" type="double"/>
		<param name="GRID_NUM" value="$(arg grid_num)" type="int"/>
		<param name="MANUAL_CROP_SIZE" value="$(arg manual_crop_size)" type="int"/>
		<!-- ray number -->
		<param name="RAY_NUM" value="180" type="int"/>
		<!-- Velodyne or 2D LiDAR -->
		<param name="IS_USE_VELODYNE" value="true" type="bool"/>
	</node>
</launch>
